scripts/ssm-conn.sh

```diff
@@ -2,49 +2,88 @@
 
 usage () {
     echo "Connects a container to a local port, for example to allow you to 'curl' requests from your local machine to the instance."
-    echo "Usage: $(basename ${0}) local_port_number [port_number] [cluster_name] [service_name] [task_name] [container_name]"
+    echo "Usage: $(basename ${0}) local_port_number -p [port_number] -c [cluster_name] -s [service_name] -t [task_name] -d [docker_container_name]"
     exit 1
 }
 
-if [[ "${#}" -lt 1 ]]; then
+while [ $OPTIND -le "$#" ]
+do
+    if getopts p:c:s:t:d: opt
+    then
+        case $opt in
+            p) port_number=$OPTARG
+            ;;
+            c) cluster_name=$OPTARG
+            ;;
+            s) service_name=$OPTARG
+            ;;
+            t) task_name=$OPTARG
+            ;;
+            d) docker_container_name=$OPTARG
+            ;;
+            \?) echo "Invalid option -$OPTARG" >&2
+            exit 1
+            ;;
+        esac
+    else
+        script_args+=("${!OPTIND}")
+        ((OPTIND++))
+    fi
+done
+
+if [[ "${#script_args[@]}" -ne 1 ]]; then
     echo "Invalid number of arguments - local port number must be supplied"
     usage
 fi
 
-if [ -z "${2}" ]; then
+
+if [ -z "${port_number}" ]; then
     port_number='8080'
-else
-    port_number=${2}
 fi
 
-if [ -z "${3}" ]; then
+if [ -z "${cluster_name}" ]; then
     cluster_name='post-award-test'
-else
-    cluster_name=${3}
 fi
 
-if [ -z "${4}" ]; then
+if [ -z "${service_name}" ]; then
     service_name='data-store'
-else
-    service_name=${4}
 fi
 
-if [ -z "${5}" ]; then
+if [ -z "${task_name}" ]; then
     task_name='post-award-test'
-else
-    task_name=${5}
 fi
 
-if [ -z "${6}" ]; then
-    container_name='data-store'
-else
-    container_name=${6}
+if [ -z "${docker_container_name}" ]; then
+    docker_container_name='data-store'
```

Comment:
Can we just call this 'container_name' docker doesn't really make sense once we're on AWS?

Reply:
I've called it 'docker container name' because that matches with the -d parameter pass. I can't use -c twice, and changing -d to match with docker container name seemed like the better option than trying to change cluster

Can keep it as container_name here in the code, but then the -d doesn't make much sense. Happy to accept suggestions!